Demo/Demo/ViewModels/PayPalWebViewModel.swift

```diff
@@ -10,15 +10,40 @@ class PayPalWebViewModel: ObservableObject, PayPalWebCheckoutDelegate {
 
     let configManager = CoreConfigManager(domain: "PayPalWeb Payments")
 
-    func createOrder(amount: String, selectedMerchantIntegration: MerchantIntegration, intent: String) async throws {
+    func createOrder(
+        amount: String,
+        selectedMerchantIntegration: MerchantIntegration,
+        intent: String,
+        shouldVault: Bool
+    ) async throws {
         // might need to pass in payee as payee object or as auth header
 
         let amountRequest = Amount(currencyCode: "USD", value: amount)
         // TODO: might need to pass in payee as payee object or as auth header
+
+        var vaultPayPalPaymentSource: VaultPayPalPaymentSource?
+        if shouldVault {
+            let attributes = Attributes(
+                vault: Vault(
+                    storeInVault: "ON_SUCCESS",
+                    usageType: "MERCHANT",
+                    customerType: "CONSUMER"
+                )
+            )
+            let paypal = VaultPayPal(attributes: attributes, experienceContext: ExperienceContext(returnURL: "https://example.com/returnUrl", cancelURL: "https://example.com/cancelUrl"))
```

Comment:
In your video, the redirect back to the Demo app looked fine, even though you would think the experienceContext should be a URL scheme to the app. What happens if the merchant doesn't set these URLs?

Reply:
The endpoint throws an error without the experienceContext field for PayPalWeb paymentl vaulting.

Reply:
In the our SDK, when we launch the ASWebSession, we construct the redirectURLString.
So this experienceContext in create Order API is only relevant to JS SDK but the API fails nonetheless
without it.

Reply:
:sigh: That's right, this is an issue with the create-order API. It doesn't accept non-https URLs.

We should add a comment to make this issue clear in the code. That basically createOrder requires an https URL, which doesn't get used by the SDK flow at all. The merchant could put `https://bogus.com` and things will work fine.

Reply:
Yes. That might be an actual site.

Reply:
Added comments above to clarify that required experienceContext fields for PayPal vault with purchase are not used in the SDK.

Reply:
I don't suppose there is a way for use to pass a URL on our merchants behalf so they don't need to do this step?

Reply:
The createOrder will just error if that URL value isn't set or is an empty string. It also errors if the string doesn't start with `https`. Victoria - can we make a JIRA for tracking taking up this issue with the API team?

Reply:
Roger

Reply:
Created - DTPPCPSDK-1492